cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java

```diff
@@ -52,11 +67,47 @@ public HttpExecutorImpl(HttpClient client) {
         this.httpClient = client;
     }
 
+    public Organization getOrganization(URI uri) throws ApiGatewayException {
+        return fromSubSubunit(getSubSubUnitDtoWithMultipleEfforts(uri));
+    }
+
+    protected SearchResponse<Organization> query(URI uri) throws NotFoundException, FailedHttpRequestException {
+        HttpResponse<String> response = sendRequestMultipleTimes(addLanguage(uri)).get();
+        if (isSuccessful(response.statusCode())) {
+            try {
+                List<Organization> organizations = getOrganizations(response);
+                return new SearchResponse<Organization>(uri)
+                        .withHits(organizations)
+                        .withSize(getCount(response, organizations));
+            } catch (JsonProcessingException e) {
+                throw new FailedHttpRequestException(e.getMessage());
+            }
+        } else if (response.statusCode() == HTTP_NOT_FOUND) {
+            throw new NotFoundException(String.format(NOT_FOUND_MESSAGE_TEMPLATE, uri));
+        } else {
+            throw new FailedHttpRequestException(errorMessage(response));
```

Comment:
Is this error message the one that gets returned to client?

Reply:
It is beeing rewrappet in the handler, but the message is kept

Reply:
So the client will receive a status code and message from cristin (from the Response object) and not from nva if this fails?